src/util/change-types.ts

```diff
@@ -310,3 +310,57 @@ export const increaseZIndexInBatch = (
         graph.setEdgeAttribute(s, 'zIndex', z + value);
     });
 };
+
+const changeStationIntType = (
```

Comment:
This also deserves a better name. Currently it looks like changing an int station from one type to another. Maybe `makeStationType`.

Moreover, I believe it is better to just do all the checks and return the type if a change is needed. Then call `changeStationType` in the following `autoChangeStationIntType` or outside.